web-client/public/js/grnstate.js

```diff
@@ -1,3 +1,46 @@
+let currentNetwork = null;
+
+const annotateLinks = network => {
+    // TODO This duplicates logic that is done on the server side for an .xlsx spreadsheet.
+    //      Think of a way to consolidate it. Having discovered this, it seems like this should
+    //      be done on the client side because it rearranges data redundantly, for ease of display.
+    network.positiveWeights = [];
+    network.negativeWeights = [];
+
+    network.links.forEach(link => {
+        if (network.sheetType === "unweighted" && !link.value) {
+            link.value = 1;
+        }
+
+        if (link.value > 0) {
+            link.type = "arrowhead";
+            // link.stroke = "MediumVioletRed";   // GRNsight v1 magenta edge color
+            link.stroke = "rgb(195, 61, 61)";     // Node coloring-consistent red edge color
+            network.positiveWeights.push(link.value);
+        } else {
+            link.type = "repressor";
+            // link.stroke = "DarkTurquoise";     // GRNsight v1 cyan edge color
+            link.stroke = "rgb(51, 124, 183)";    // Node coloring-consistent blue edge color
+            network.negativeWeights.push(link.value);
+        }
+    });
+};
+
 export const grnState = {
-    dashedLine: false
+    name: null,
+
+    newNetwork: false,
+
+    get network () {
+        return currentNetwork;
+    },
+
+    set network (network) {
+        currentNetwork = network;
+        this.newNetwork = true;
+    },
```

Comment:
Here is the start of a getter and setter for the `network` property. I expect that the setter will grow more as additional code is refactored.

I was momentarily tempted to just integrate `annotateLinks` here. But if you look at the original code, `annotateLinks` is only called upon _import_ and not _open_. To play it safe I maintained this separation. However, you might discover that the operation is equally valid both ways, and if you do, then calling `annotateLinks` from with the setter can potentially streamline things.